### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by davesnx**
- Add examples of all the options
- Explain the context of why
- Adding use-cases
- Add appvoyer label
- Add learning jq section

PS: The only part that I miss here is the Contribution section, but for now I woudn't add it now.

Taking a look into the changes is useless, check https://github.com/sanack/node-jq/tree/Improve_Documentation

Solves https://github.com/sanack/node-jq/issues/34

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,11 +31,14 @@ npm install node-jq --save
 
 ## Usage
 
-Usually in your CLI with jq you run:
+#### jq example
```
Comment (davesnx) [inline README.md]: isn't that line redundant?

**Comment by davesnx** [comment]
Do you think is worth it to explain this issues https://github.com/sanack/node-jq/issues/52.
In the case that you are running a filter that changes the output, the options of ouput doesn't make sense.

File: README.md (hunk 1 above)
Comment (mackermans) [inline README.md]: Yeah I'm not completely sure, I wanted to make the difference between `jq` and `node-jq` usage more obvious, that's why I added it.

Comment (davesnx) [inline README.md]: ye, I catch the intention

**Comment by davesnx** [comment]
restest this please

**Pull request merged**